src/PIL/PpmImagePlugin.py

```diff
@@ -123,7 +128,176 @@ def _open(self):
+        half_token = False
+        tokens_read = 0
+        while True:
+            block = self._read_block()  # read next block
+            if not block:
+                if token_spans:
+                    block = bytearray(b" ")  # flush half_token
+                else:
+                    raise ValueError("Reached EOF while reading data")
+
+            while block and comment_spans:
+                comment_end = self._find_comment_end(block)
+                if comment_end != -1:  # comment ends in this block
+                    block = block[comment_end + 1 :]  # delete tail of previous comment
+                    break
+                else:  # comment spans whole block
+                    block = self._read_block()
+
+            block, comment_spans = self._ignore_comments(block)
+
+            if token_spans:
+                block = half_token + block  # stitch half_token to new block
+                token_spans = False
+
+            tokens = block.split()
+
+            if block and not block[-1:].isspace():  # block might split token
+                token_spans = True
+                half_token = tokens.pop()  # save half token for later
+                if len(half_token) > max_len:  # prevent buildup of half_token
+                    raise ValueError(
+                        f"Token too long found in data: {half_token[:max_len + 1]}"
+                    )
+
+            for token in tokens:
+                if len(token) > max_len:
+                    raise ValueError(
+                        f"Token too long found in data: {token[:max_len + 1]}"
+                    )
+                try:
+                    token = int(token)
+                except ValueError:
+                    raise ValueError(
+                        f"Non-decimal-ASCII found in data: {token}"
+                    ) from None
+                tokens_read += 1
+                if token > maxval:
+                    raise ValueError(f"Channel value too large for this mode: {token}")
+                decoded_data += token.to_bytes(bytes_per_sample, "big")
+                if tokens_read == total_tokens:  # finished!
+                    return decoded_data
+
+    def decode(self, buffer):
+        self.size = self.state.xsize * self.state.ysize
+        rawmode = self.args[0]
+
+        if self.mode == "1":
+            decoded_data = self._decode_bitonal()
+            rawmode = "1;8"
+        elif self.mode == "L":
+            decoded_data = self._decode_blocks(channels=1, depth=8)
+        elif self.mode == "I":
+            if rawmode == "I;16B":
+                decoded_data = self._decode_blocks(channels=1, depth=16)
+            elif rawmode == "I;32B":
+                decoded_data = self._decode_blocks(channels=1, depth=32)
+        elif self.mode == "RGB":
+            decoded_data = self._decode_blocks(channels=3, depth=8)
+
+        self.set_as_raw(bytes(decoded_data), rawmode)
+        return -1, 0
```

Comment:
```suggestion
        return 0, 0
```
I think this should be zero, as per https://github.com/python-pillow/Pillow/blob/180c89f7559c7faeddd1a215e2a2937dc3afb4d9/src/PIL/ImageFile.py#L668-L674